app/com/linkedin/drelephant/tuning/AbstractFitnessManager.java

```diff
@@ -325,15 +325,19 @@ public final boolean execute() {
     }
     logger.info(
         "Disable Tuning if Required . Disabled based on number of iteration reached or other terminating conditions for algorithms");
-    if (tuningJobExecutionParamSet != null && tuningJobExecutionParamSet.size() >= 1) {
-      Set<JobDefinition> jobDefinitionSet = new HashSet<JobDefinition>();
-      for (TuningJobExecutionParamSet completedJobExecutionParamSet : tuningJobExecutionParamSet) {
-        JobDefinition jobDefinition = completedJobExecutionParamSet.jobSuggestedParamSet.jobDefinition;
-        if (isTuningEnabled(jobDefinition.id)) {
-          jobDefinitionSet.add(jobDefinition);
+    try {
+      if (tuningJobExecutionParamSet != null && tuningJobExecutionParamSet.size() >= 1) {
+        Set<JobDefinition> jobDefinitionSet = new HashSet<JobDefinition>();
+        for (TuningJobExecutionParamSet completedJobExecutionParamSet : tuningJobExecutionParamSet) {
+          JobDefinition jobDefinition = completedJobExecutionParamSet.jobSuggestedParamSet.jobDefinition;
+          if (isTuningEnabled(jobDefinition.id)) {
+            jobDefinitionSet.add(jobDefinition);
+          }
         }
+        checkToDisableTuning(jobDefinitionSet);
       }
-      checkToDisableTuning(jobDefinitionSet);
+    }catch (Exception e){
```

Comment:
@erwa 
Hi Anthony ,
This is temporary to actually find out the exception and its reason.  Once we know the exact reason of exception , then will handle it and will have specific exception

Reply:
Can you check the Dr. Elephant logs to figure this out?

Reply:
Dr elephant logs doesn't have information . The last log related to this was "Disable Tuning if Required . Disabled based on number of iteration reached or other terminating conditions for algorithms"  but no further log was there in the corresponding thread. 

Catching Exceptions in the loop is required to take care of intermittent failure of DBs .Code should not come out of loop , even if there are failures while updating job definitions of one of the jobs (it might be because of optimisiticlockexceptions) , it should continue for other jobs .

Reply:
Figured out the reason for failure , it was failing because of optimisticlockexception because of data (removed the data to not have exception). However as said earlier , it is better to have catching exception inside loop , so that loop should complete even if there are run time exceptions for some of the job